### DISCUSSION TITLE
PylonCapture lags when combined with PredictPose(DLC)

### DISCUSSION THREAD

**Discussion by trose-neuro** [bonsai-rx/bonsai#1233] [Q&A] [answered]
Hi all,

we have trouble with lagging recording from the PylonCapture node when using DLC live. X-posting from the DLClive discussions:
https://github.com/bonsai-rx/deeplabcut/issues/16

We had this on discord a while back as well.

Are there any updates on this? The delay is crippling.

best
Tobias

**Comment by bruno-f-cruz** [comment, marked as answer]
Have you tried decreasing frame rate and image resolution?

**Reply by trose-neuro** [reply]
Does not change problem. Also: CamareCapture Node works but does not maintain pylon settings (see link to original post).

**Reply by bruno-f-cruz** [reply]
That's bizarre... how low of a FPS and resolution have you tried? If you do 100x100 and 5fps, for instance, it still lags? 
Is the GPU being recognized by the runtime or is it running with CPU?

**Reply by trose-neuro** [reply]
Yes - it is indeed weird. I don't hink we went that low with the resolution. Can try. And yes: it is running on GPU (RTX3070Ti). Also weird: it runs without any lag at high res 700x700@60Hz with the CameraCapture node. If you check discord, there has also been another user with the same issue (follow my last post).

Using camera capture is a possibility - sadly presetting everything in pylon (triggering exposure etc) does not work for cropping (ROI settings). Hence our interest to continue to use the Pylon node.

**Reply by bruno-f-cruz** [reply]
I am not sure what you are doing with CameraCapture, but this node drops frames if the runtime cant keep up, so that would explain why you have no lags...

**Reply by trose-neuro** [reply]
We will try to get into this a bit more systematically now. Will report back when we know more.

**Reply by trose-neuro** [reply]
To follow up on this: Turns out this indeed was a simple GPU load issue. Binning on the camera solved this.

**Comment by jfrazao** [comment]
Can you track with DLC at a lower frame rate that you capture and save?
If you can try this:

 CaptureVideo->VideoWriter->ObserveOn(DefaultScheduler)->Slice(step:10)->DLC()
if not try just this:
     CaptureVideo->VideoWriter->ObserveOn(DefaultScheduler)->DLC()

**Reply by bruno-f-cruz** [reply]
You need to be careful with this one:
`CaptureVideo->VideoWriter->ObserveOn(DefaultScheduler)->DLC()
`
Yes, it will uncouple the two processes, but you will still queue the frames to the DLC pipeline. Given the resolution of the frames I bet memory will blow up pretty quickly :P

**Reply by trose-neuro** [reply]
Thanks. Will try this! (sadly, this got pushed a bit down my priority list. Will come back).
